File: py/src/images/stability.py
import boto3
import json
import base64

client = boto3.client(service_name='bedrock-runtime', region_name="us-west-2")

stability_image_config = json.dumps({
    "prompt": "a photo of a dragon"
})

response = client.invoke_model(
    body=stability_image_config, 
    modelId="stability.stable-image-core-v1:1", 
    accept="application/json", 
    contentType="application/json")

response_body = json.loads(response.get("body").read())
base64_image = response_body.get("images")[0]

base_64_image = base64.b64decode(base64_image)

file_path = "dragon.png"
with open(file_path, "wb") as f:
    f.write(base_64_image)

File: ts/src/images/stability.ts
import { BedrockRuntimeClient, InvokeModelCommand } from '@aws-sdk/client-bedrock-runtime'
import { writeFileSync } from 'fs'

const client = new BedrockRuntimeClient({ region: 'us-west-2' })

const stabilityImageConfig = {
    prompt: 'a photo of a dragon'
}

async function invokeModel() {
    const response = await client.send(new InvokeModelCommand({
        modelId: 'stability.stable-image-core-v1:1',
        body: JSON.stringify(stabilityImageConfig),
        accept: 'application/json',
        contentType: 'application/json'
    }));
    const responseBody = JSON.parse(new TextDecoder().decode(response.body));
    saveImage(responseBody.images[0], 'dragonDiffusion.png');
}

function saveImage(base64Data: string, fileName: string) {
    const imageBuffer = Buffer.from(base64Data, 'base64');
    writeFileSync(fileName, imageBuffer);
}

invokeModel();
